core/templates/domain/collection/collection-update.service.ts

```diff
@@ -19,243 +19,266 @@
+        exploration_id: explorationId
+      }, (changeDict, collection) => {
+        // Apply.
+        let explorationId = this._getExplorationIdFromChangeDict(changeDict);
+        collection.deleteCollectionNode(explorationId);
+      }, (changeDict, collection) => {
+        // Undo.
+        collection.addCollectionNode(oldCollectionNode);
+      });
+  }
 
-      /**
-       * Changes the title of a collection and records the change in the
-       * undo/redo service.
-       */
-      setCollectionTitle: function(collection, title) {
-        var oldTitle = angular.copy(collection.getTitle());
-        _applyPropertyChange(
-          collection, COLLECTION_PROPERTY_TITLE, title, oldTitle,
-          function(changeDict, collection) {
-            // ---- Apply ----
-            var title = _getNewPropertyValueFromChangeDict(changeDict);
-            collection.setTitle(title);
-          }, function(changeDict, collection) {
-            // ---- Undo ----
-            collection.setTitle(oldTitle);
-          });
-      },
+  /**
+   * Changes the title of a collection and records the change in the
+   * undo/redo service.
+   */
+  setCollectionTitle(
+      collection: Collection,
+      title: string): void {
+    let oldTitle = collection.getTitle();
+    this._applyPropertyChange(
+      collection,
+      CollectionDomainConstants.COLLECTION_PROPERTY_TITLE, title, oldTitle,
+      (changeDict, collection) => {
+        // ---- Apply ----
+        let title = this._getNewPropertyValueFromChangeDict(changeDict);
+        collection.setTitle(title);
+      }, (changeDict, collection) => {
+        // ---- Undo ----
+        collection.setTitle(oldTitle);
+      });
+  }
 
-      /**
-       * Changes the category of a collection and records the change in the
-       * undo/redo service.
-       */
-      setCollectionCategory: function(collection, category) {
-        var oldCategory = angular.copy(collection.getCategory());
-        _applyPropertyChange(
-          collection, COLLECTION_PROPERTY_CATEGORY, category, oldCategory,
-          function(changeDict, collection) {
-            // Apply.
-            var category = _getNewPropertyValueFromChangeDict(changeDict);
-            collection.setCategory(category);
-          }, function(changeDict, collection) {
-            // Undo.
-            collection.setCategory(oldCategory);
-          });
-      },
+  /**
+   * Changes the category of a collection and records the change in the
+   * undo/redo service.
+   */
+  setCollectionCategory(
+      collection: Collection,
+      category: string): void {
+    let oldCategory = collection.getCategory();
+    this._applyPropertyChange(
+      collection,
+      CollectionDomainConstants.COLLECTION_PROPERTY_CATEGORY, category,
+      oldCategory, (changeDict, collection) => {
+        // Apply.
+        let category = this._getNewPropertyValueFromChangeDict(changeDict);
```

Comment:
Can the name of the variable "category" be changed to something like "newCategory", since there is already a variable named category in the parameters. @savithakj wdyut?

Reply:
updated